### ISSUE TITLE
offset without limit

### ISSUE THREAD

**Issue by Dushistov** [couchbase/couchbase-lite-core#865]
I try to create query with `OFFSET`, but without `LIMIT`.
```
 {"WHAT": ["._id"], "WHERE": ["=", [".type"], "MyDoc"], "ORDER_BY": [["DESC", [".departure_time"]], \
["ASC", ["._id"]]],"OFFSET": 1}
```
couchbase-lite-core converts it to 
```
SELECT fl_result(_doc.key) FROM kv_default AS _doc WHERE (fl_value(_doc.body, 'type') = \
'MyDoc') AND (_doc.flags & 1 = 0) ORDER BY fl_value(_doc.body, 'departure_time')
```

But this is invalid `SQL` from `sqlite` point of view.
Normal way to fix this for `sqlite` is adding `LIMIT -1` before `OFFSET`,
but  after  #329  was fixed it is impossible `LIMIT -1` is converted to `LIMIT MAX(0, -1)` and not works as expected.

**Comment by borrrden** [comment]
Well that is somewhat of a pain....in the meantime the workaround would be to set the limit to something high (like int32 max).  Are you sure you've pasted the entire SQLite statement though?  I don't see an offset in there anywhere.

**Comment by Dushistov** [comment]
@borrrden 

I copy/paste the first line of error, here full error message:
```
(cblite) query {"WHAT": ["._id"], "WHERE": ["=", [".type"], "MyDoc"], "ORDER_BY": [["DESC", [".departure_time"]], ["ASC", ["._id"]]],"OFFSET": 1}
09:55:14.069298| [DB] ERROR: SQLite error (code 1): near "OFFSET": syntax error in "SELECT fl_result(_doc.key) FROM kv_default AS _doc WHERE (fl_value(_doc.body, 'type') = 'MyDoc') AND (_doc.flags & 1 = 0) ORDER BY fl_value(_doc.body, 'departure_time') DESC, _d
09:55:14.069376| [SQL] ERROR: near "OFFSET": syntax error (1/1)
09:55:14.069419| [DB] WARNING: {DB#1} SQLite error compiling statement "SELECT fl_result(_doc.key) FROM kv_default AS _doc WHERE (fl_value(_doc.body, 'type') = 'MyDoc') AND (_doc.flags & 1 = 0) ORDER BY fl_value(_doc.body, 'departure_time') DESC, _doc.key ASC OFFSET MAX(0, 1)": near "OFFSET": syntax error
Error compiling query: near "OFFSET": syntax error (3/1)
(cblite) 
```

**Comment by borrrden** [comment]
Migrated to https://issues.couchbase.com/browse/CBL-529

**Comment by snej** [comment]
Fixed in commit 576efe30b887331a646ccc67be43ebc2cf4891bd.
